web3/_utils/module_testing/eth_module.py

```diff
@@ -1405,6 +1405,74 @@ def test_async_provider_default_block(
         # reset to default
         async_w3.eth.default_block = "latest"
 
+    @pytest.mark.asyncio
+    async def test_async_eth_new_filter(self, async_w3: "Web3") -> None:
+        filter = await async_w3.eth.filter({})
+
+        changes = await async_w3.eth.get_filter_changes(
+            filter.filter_id
+        )  # type: ignore
+        assert is_list_like(changes)
+        assert not changes
+
+        logs = await async_w3.eth.get_filter_logs(filter.filter_id)  # type: ignore
+        assert is_list_like(logs)
+        assert not logs
+
+        result = await async_w3.eth.uninstall_filter(filter.filter_id)  # type: ignore
+        assert result is True
+
+    @pytest.mark.asyncio
+    async def test_async_eth_new_block_filter(self, async_w3: "Web3") -> None:
+        filter = await async_w3.eth.filter("latest")
+        assert is_string(filter.filter_id)
+
+        changes = await async_w3.eth.get_filter_changes(
+            filter.filter_id
+        )  # type: ignore
+        assert is_list_like(changes)
+        assert not changes
+
+        # TODO: figure out why this fails in go-ethereum
```

Comment:
We can take these TODOs out. Looks like it fails right [here](https://github.com/ethereum/go-ethereum/blob/b196ad1c165ecd6c9edaca520e7161a58e50eb06/eth/filters/api.go#L367). If I'm reading this correctly, neither a `BlocksSubscription` (created via `w3.eth.filter("latest")`) nor a `PendingTransactionsSubscription` (created via `w3.eth.filter("pending")`) is accepted for `eth_getLogs`. Only a `LogsSubscription` (or `eth_newFilter`... created by supplying the `filter_params` dict) is supported and we have tests for that below.

We can take these out from the sync tests as well.

Reply:
Removed!